### ISSUE TITLE
[BUG] When a non-root user views the edit page of a piece of content, the Delete button doesn't appear

### ISSUE THREAD

**Issue by use** [bolt/bolt#6068]
Non-root users who view the edit page of a piece of content and have the delete permission for that contenttype, don't see a Delete button, even though they can delete the content by other means (such as via the dropdown arrow and checkboxes on the content list view).

Additionally, an error appears in the logs whenever such a visitor views the edit page.

`Info	authentication	editcontent	/bolt/editcontent/pages/1	Configuration error: delete is not granted to any roles.
bolt\vendor\bolt\bolt\src\AccessControl\Permissions.php line: 315
`

Details
-------

 - Relevant Bolt Version: [ 3.0.11, 3.1.5, 3.2.2 ]
 - Install type: [ Composer install ]
 - PHP version: [ 7.0 ]
 - Used web server: [ Apache ]

Reproduction
------------
​
 1. **Steps to reproduce**:
    * Grant delete permissions to a role
    * Assign a user to that role
    * Login as that user
    * When viewing a list of content, and clicking the dropdown arrow, the Delete options appears as expected
    * Click to edit a piece of content
    * The log entry mentioned above now appears in the System Log
    * The user does not see the Delete button
 2. **Expected result**: 
    * The Delete button appears (screenshot for root user)
![image](https://cloud.githubusercontent.com/assets/94230/20502948/4abcc514-aff5-11e6-9f03-320fe69cedfe.png)

 3. **Actual result**: 
    * The Delete button does not appear (screenshot for non-root user who has delete permissions) 
![image](https://cloud.githubusercontent.com/assets/94230/20502987/6ee0235a-aff5-11e6-9c62-711fd5b9a732.png)

**Comment by SofaCitizen** [comment]
I had this issue also with a new site build. I worked around it by adding a permission entry to the global array for "delete" which then stopped the error messages and displayed the delete button.

Obviously, that is not a correct fix but can be used as a workaround in the meantime.

**Comment by use** [comment]
I just revisited this - for anyone else having this problem, you can do like @SofaCitizen says and add a global delete permission like so:

```` yml
# permissions.yml
global: # don't add this line, it's already there
    delete: [admin, developer, chief-editor] # put a line like this anywhere under the global section
````

Looks like this workaround is needed in 3.2.7 as well.

**Comment by bobdenotter** [comment]
Thanks, @SofaCitizen, @use,

> Obviously, that is not a correct fix but can be used as a workaround in the meantime.

Actually, I think it is. "root" is always allowed to do that, but others need permission. It makes sense to add that line, because it's aparently missing. 👍

**Comment by GawainLynch** [comment]
For anyone looking at this in future tracking down similar problems. Please see the latter conversation in #6457 and the fix in #6468, as the suggestions here are actually quite wrong and will hid larger permissions problems in their worse case.
